### PR TITLE
added Alternative reach access

### PR DESCRIPTION
**Pull request by LoadCake**
Added support for making baur sneeze with blaze and flash

### CODE REVIEW

File: projects/Randomizer/game/behaviour_changes/alternative_baurs_reach_access.cpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    };
+    bool should_open_condition;
+
+    IL2CPP_INTERCEPT(BaurEntity, void, ResolveDamage, (app::BaurEntity * this_ptr, app::DamageResult result)){
+        should_open_condition = false;
+        if(ability_should_open.contains(result.Damage->fields.m_abilityType)) {
+            should_open_condition = ability_should_open[result.Damage->fields.m_abilityType].get<bool>();
+        }
+
+        if(should_open_condition) {
```
Comment (timoschwarzer) [inline projects/Randomizer/game/behaviour_changes/alternative_baurs_reach_access.cpp:24]: space

File: projects/Randomizer/game/behaviour_changes/alternative_baurs_reach_access.cpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+#include <Modloader/interception_macros.h>
+
+
+namespace {
+
+    std::unordered_map<app::AbilityType__Enum, core::api::uber_states::UberState> ability_should_open {
+        {app::AbilityType__Enum::Blaze, core::api::uber_states::UberState(6, 420)},
+        {app::AbilityType__Enum::GlowSpell, core::api::uber_states::UberState(6, 421)},
+    };
+    bool should_open_condition;
```
Comment (timoschwarzer) [inline projects/Randomizer/game/behaviour_changes/alternative_baurs_reach_access.cpp:16]: this does not need to be in global scope

File: projects/Randomizer/game/behaviour_changes/alternative_baurs_reach_access.cpp [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                std::vector<std::string>{
+                    "interactives",
+                    "baurSetup",
+                    "attackRestrictZone"}
+            );
+
+            if (il2cpp::unity::is_valid(ability_restrict_zone_go)) {
+                restrict_zone = il2cpp::WeakGCRef(il2cpp::unity::get_component<app::SeinAbilityRestrictZone>(ability_restrict_zone_go, types::SeinAbilityRestrictZone::get_class()));
+                restrict_zone_effect = core::reactivity::watch_effect([] {
+                    if(restrict_zone->is_valid()) {
```
Comment (timoschwarzer) [inline projects/Randomizer/game/behaviour_changes/alternative_baurs_reach_access.cpp:52]: space

File: projects/Randomizer/game/behaviour_changes/alternative_baurs_reach_access.cpp [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+                        auto mask = static_cast<int>((**restrict_zone)->fields.RestrictMask);
+                         if (ability_should_open[app::AbilityType__Enum::Blaze].get<bool>()) {
+                             mask = mask & ~static_cast<int>(app::SeinAbilityRestrictZoneMask__Enum::Attack);
+                         }
+                        (**restrict_zone)->fields.RestrictMask = static_cast<app::SeinAbilityRestrictZoneMask__Enum>(mask);
```
Comment (timoschwarzer) [inline projects/Randomizer/game/behaviour_changes/alternative_baurs_reach_access.cpp:57]: indentation

**Review by timoschwarzer — CHANGES_REQUESTED** [review]

File: projects/Randomizer/game/behaviour_changes/alternative_baurs_reach_access.cpp [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+    IL2CPP_INTERCEPT(BaurEntity, void, ResolveDamage, (app::BaurEntity * this_ptr, app::DamageResult result)){
+        bool should_open_condition = false;
+        if (ability_should_open.contains(result.Damage->fields.m_abilityType)) {
+            should_open_condition = ability_should_open.at(result.Damage->fields.m_abilityType).get<bool>();
+        }
+
+        if (should_open_condition) {
+            auto r = result.Damage->fields.m_damageType = app::DamageType__Enum::Wind;
+        }
```
Comment (timoschwarzer) [inline projects/Randomizer/game/behaviour_changes/alternative_baurs_reach_access.cpp:25]:
```suggestion
        const auto should_open =
            ability_should_open.contains(result.Damage->fields.m_abilityType) &&
            ability_should_open.at(result.Damage->fields.m_abilityType).get<bool>();
        
        if (should_open) {
            result.Damage->fields.m_damageType = app::DamageType__Enum::Wind;
        }
```

File: projects/Randomizer/game/behaviour_changes/alternative_baurs_reach_access.cpp [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+            );
+
+            if (il2cpp::unity::is_valid(ability_restrict_zone_go)) {
+                restrict_zone = il2cpp::WeakGCRef(il2cpp::unity::get_component<app::SeinAbilityRestrictZone>(ability_restrict_zone_go, types::SeinAbilityRestrictZone::get_class()));
+                restrict_zone_effect = core::reactivity::watch_effect([] {
+                    if (restrict_zone->is_valid()) {
+                        auto mask = static_cast<int>((**restrict_zone)->fields.RestrictMask);
+                        if (ability_should_open.at(app::AbilityType__Enum::Blaze).get<bool>()) {
+                            mask = mask & ~static_cast<int>(app::SeinAbilityRestrictZoneMask__Enum::Attack);
+                        }
```
Comment (timoschwarzer) [inline projects/Randomizer/game/behaviour_changes/alternative_baurs_reach_access.cpp:55]: handling for Flash is missing

File: projects/Randomizer/uber_states/uber_state_initialization.cpp [hunk 7]
```diff
@@ -338,6 +338,8 @@ namespace randomizer {
                 add_state<app::SerializedBooleanUberState>(UberStateGroup::RandoState, "rainLiftedInInkwater", 401, false),
                 add_state<app::SerializedBooleanUberState>(UberStateGroup::RandoState, "regenTreeDrained", 402, false),
                 add_state<app::SerializedBooleanUberState>(UberStateGroup::RandoState, "cleanWater", 2000, false),
+                add_state<app::SerializedBooleanUberState>(UberStateGroup::RandoState, "BaurSneezeWithBlaze", 420, false),
+                add_state<app::SerializedBooleanUberState>(UberStateGroup::RandoState, "BaurSneezeWithFlash", 421, false),
```
Comment (timoschwarzer) [inline projects/Randomizer/uber_states/uber_state_initialization.cpp:342]: B → b and move to RandoConfig

**Review by timoschwarzer — CHANGES_REQUESTED** [review]

File: projects/Randomizer/game/behaviour_changes/alternative_baurs_reach_access.cpp (hunk 6 above)
Comment (LoadCake) [reply projects/Randomizer/game/behaviour_changes/alternative_baurs_reach_access.cpp:55]: Flash does not get locked by the Damage mask. Even with the original restrict zone it was possible to use flash

**Review by timoschwarzer — APPROVED** [review]

**Pull request merged**